pkg/sdk/sdk_test.go

```diff
@@ -0,0 +1,390 @@
+						log.Error(err.Error())
+						return
+					}
+					data.Id = id
+
+					response.ToJson(w, http.StatusOK, "", nil)
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err.Error())
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: GeneratePdf :: id required",
+			id:   "",
+			data: map[string]interface{}{"id": "1"},
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/generate/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					vars := mux.Vars(r)
+					//we take id as a parameter from url path
+					id, ok := vars["id"]
+					if !ok {
+						log.Error("here")
+						response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrIdNeeded), nil)
+						return
+					}
+					var data modelV.GenerateReq
+					err := json.NewDecoder(r.Body).Decode(&data)
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrDecodingData), nil)
+						log.Error(err.Error())
+						return
+					}
+					data.Id = id
+
+					response.ToJson(w, http.StatusOK, "", nil)
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != errors.New("non success status code received : 404").Error() {
+					t.Errorf("Want: %v, Got: %v", "non success status code received : 404", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: GeneratePdf :: incorrect status code",
+			id:   "1",
+			data: map[string]interface{}{"id": "1"},
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/generate/{id}", http.MethodPost, func(w http.ResponseWriter, r *http.Request) {
+					vars := mux.Vars(r)
+					//we take id as a parameter from url path
+					id, ok := vars["id"]
+					if !ok {
+						log.Error("here")
+						response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrIdNeeded), nil)
+						return
+					}
+					var data modelV.GenerateReq
+					err := json.NewDecoder(r.Body).Decode(&data)
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrDecodingData), nil)
+						log.Error(err.Error())
+						return
+					}
+					data.Id = id
+
+					response.ToJson(w, http.StatusNotFound, "", nil)
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if !reflect.DeepEqual(err.Error(), errors.New("non success status code received : 404").Error()) {
+					t.Errorf("Want: %v, Got: %v", "non success status code received : 404", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			svr := tt.setupFunc()
+			defer tt.cleanupFunc(svr)
+
+			calls := NewHtmlToPdfSvc(svr.URL)
+			_, err := calls.GeneratePdf(tt.data, tt.id)
```

Comment:
This sdk function returns error and the file bytes. The file bytes must also be validated and must be the same as returned by the server.

Reply:
done